File: 0932-monotonic-array/0932-monotonic-array.java
class Solution {
    public boolean isMonotonic(int[] nums) {
        int l=0;
        while(l<nums.length-1 && nums[l]==nums[l+1]) l++;
        if(l==nums.length-1) return true;
        if(nums[l]<nums[l+1]){
            for(int i=l+1; i<nums.length-1;i++){
                if(nums[i]>nums[i+1]) return false;
            }
        }else{
            for(int i=l+1; i<nums.length-1;i++){
                if(nums[i]<nums[i+1]) return false;
            }
        }

        return true;
    }
}

File: 0932-monotonic-array/0932-monotonic-array.cpp
class Solution {
public:
    bool isMonotonic(vector<int>& nums) {
        int n = nums.size();
        if (n <= 2)
            return true;

        int i = 0;
        while (i + 1 < n && nums[i] == nums[i + 1]) i++;
        if (i + 1 < n && nums[i + 1] > nums[i]) {
            while (i + 1 < n) {
                if (nums[i + 1] < nums[i]) return false;
                i++;
            }
        } else {
            while (i + 1 < n) {
                if (nums[i + 1] > nums[i]) return false;
                i++;
            }
        }
        return true;
    }
};